src/linear_algebra/dual.lean

```diff
@@ -886,6 +898,37 @@ begin
   exact finrank_range_add_finrank_ker f,
 end
 
+open linear_map
+/-- `T` is surjective if and only if `T.dual_map` is injective -/
+lemma linear_map.surjective_iff_dual_injective {V W : Type*} [add_comm_group V] [add_comm_group W]
+  [module K V] [module K W] [finite_dimensional K W]
+  (T : V →ₗ[K] W) : function.surjective T ↔ function.injective T.dual_map :=
+begin
+  refine ⟨λ h, linear_map.dual_map_injective_of_surjective T h, _⟩,
+  rw [← range_eq_top, ← ker_eq_bot],
+  intro h,
+  apply finite_dimensional.eq_top_of_finrank_eq,
+  rw ← finrank_eq_zero at h,
+  rw [← add_zero (finite_dimensional.finrank K T.range), ← h,
+      ← linear_map.finrank_range_dual_map_eq_finrank_range,
+      linear_map.finrank_range_add_finrank_ker, subspace.dual_finrank_eq],
+end
+
+/-- `T` is injective if and only if `T.dual_map` is surjective -/
+lemma linear_map.injective_iff_dual_surjective {V W : Type*} [add_comm_group V] [add_comm_group W]
+  [module K V] [module K W] [finite_dimensional K V] [finite_dimensional K W]
+  (T : V →ₗ[K] W) : function.injective T ↔ function.surjective T.dual_map :=
```

Comment:
I suspect one direction of this lemma is also true without needing finite dimensionality; probably the one with `function.surjective T.dual_map` as an assumption.

Reply:
Yeah, you're probably right (I'll try this later at some point). The finite-dimensional proof seems to be far more straightforward though.

Reply:
You're right, this isn't as easy as I thought. No need to do it in this PR.